test/parallel/test-runner-coverage.js

```diff
@@ -154,3 +140,56 @@ test('test spec coverage reporter', async (t) => {
     assert(!findCoverageFileForPid(result.pid));
   });
 });
+
+test('single process coverage is the same with --test', () => {
+  if (!process.features.inspector) {
+    return;
+  }
+
+  const fixture = fixtures.path('test-runner', 'coverage.js');
+  const args = [
+    '--test', '--experimental-test-coverage', '--test-reporter', 'tap', fixture,
+  ];
+  const result = spawnSync(process.execPath, args);
+  const report = getTapCoverageFixtureReport();
+
+  assert(result.stdout.toString().includes(report));
+  assert.strictEqual(result.stderr.toString(), '');
+  assert.strictEqual(result.status, 0);
+  assert(!findCoverageFileForPid(result.pid));
+});
+
+test('coverage is combined for multiple processes', () => {
+  if (!process.features.inspector) {
+    return;
+  }
+
+  let report = [
+    '# start of coverage report',
+    '# file | line % | branch % | funcs % | uncovered lines',
+    '# test/fixtures/test-runner/combined_coverage/common.js | 89.86 | ' +
+    '62.50 | 100.00 | 8, 13, 14, 18, 34, 35, 53',
+    '# test/fixtures/test-runner/combined_coverage/first.test.js | 83.33 | ' +
+    '100.00 | 50.00 | 5, 6',
+    '# test/fixtures/test-runner/combined_coverage/second.test.js | 100.00 ' +
+    '| 100.00 | 100.00 | ',
+    '# test/fixtures/test-runner/combined_coverage/third.test.js | 100.00 | ' +
+    '100.00 | 100.00 | ',
+    '# all files | 90.72 | 72.73 | 88.89 |',
+    '# end of coverage report',
+  ].join('\n');
+
+  if (common.isWindows) {
+    report = report.replaceAll('/', '\\');
+  }
+
+  const fixture = fixtures.path('test-runner', 'combined_coverage');
+  const args = [
+    '--test', '--experimental-test-coverage', '--test-reporter', 'tap', fixture,
+  ];
+  const result = spawnSync(process.execPath, args);
+
+  assert(result.stdout.toString().includes(report));
+  assert.strictEqual(result.stderr.toString(), '');
```

Comment:
```suggestion
  assert.strictEqual(result.stderr.toString(), '');
  assert(result.stdout.toString().includes(report));
```
I'v learned the hard way that this order helps understanding failures faster